From 32241d9e06b4541001361985d622175988b6f84b Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Wed, 2 Jun 2021 22:22:16 -0400
Subject: [PATCH] cinnamon-screensaver-main.py: Don't register with the session
 manager.

When cinnamon-screensaver-command is called when reacting to
logout/shutdown dialogs, cinnamon-screensaver is launched and
attempts to register with cinnamon-session, which is already
stuck waiting for cinnamon-screensaver-command to return.

There's no need to register with the session.
---
 src/cinnamon-screensaver-main.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/cinnamon-screensaver-main.py b/src/cinnamon-screensaver-main.py
index 964e862..a22c75f 100755
--- a/src/cinnamon-screensaver-main.py
+++ b/src/cinnamon-screensaver-main.py
@@ -30,8 +30,7 @@ class Main(Gtk.Application):
     """
     def __init__(self):
         super(Main, self).__init__(application_id="org.cinnamon.ScreenSaver",
-                                   register_session=True,
-                                   inactivity_timeout=5000,
+                                   inactivity_timeout=10000,
                                    flags=Gio.ApplicationFlags.IS_SERVICE)
 
         # Our service must be set up before we register with the session manager.

